File: qqtech9-backend/src/models/schedules.js
import { BusinessLogicError } from "../utils/errors.js";

export class Schedules {
    constructor(schedule) {
        this.id = schedule.id;
        this.startTime = schedule.start_time ?? schedule.startTime;
        this.endTime = schedule.end_time ?? schedule.endTime;
        this.createdAt = schedule.created_at ?? schedule.createdAt;
        this.updatedAt = schedule.updated_at ?? schedule.updatedAt;
        
        this.userId = schedule.user_id ?? schedule.userId;
        this.user = schedule.user;
    }

    static fromArray(schedules) {
        return schedules.map((schedule) => new Schedules(schedule));
    }

}

File: python-workers/models/schedules.py
from sqlalchemy import UUID, Column, Integer, String, Text, Boolean, DateTime, Time, func
from config.database import Base

class Schedules(Base):
    __tablename__ = 'schedules'

    id = Column(UUID, primary_key=True)
    user_id = Column(UUID, nullable=False)
    start_time = Column(DateTime, nullable=False)
    end_time = Column(DateTime, nullable=False)
    created_at = Column(DateTime, server_default=func.now())
    updated_at = Column(DateTime, server_default=func.now(), onupdate=func.now())
    
    
    @staticmethod
    def from_array(schedules_array):
        return [Schedules(**schedule) if isinstance(schedule, dict) else schedule for schedule in schedules_array]